looper/looper.py

```diff
@@ -555,20 +566,700 @@ def __call__(self, args, remaining_args):
+                html_file.close()
+                if warnings:
+                    _LOGGER.warn("Warning: " + ''.join(list(set(warnings))))
+
+        def create_sample_html(all_samples, sample_name, sample_stats,
+                               index_html):
+            # Produce an HTML page containing all of a sample's objects
+            # and the sample summary statistics
+            reports_dir = os.path.join(self.prj.metadata.output_dir, "reports")
+            html_filename = sample_name + ".html"
+            html_page = os.path.join(
+                reports_dir, html_filename.replace(' ', '_').lower())
+            sample_page_relpath = os.path.relpath(
+                html_page, self.prj.metadata.output_dir)
+            single_sample = all_samples[all_samples['sample_name'] == sample_name]
+            if not os.path.exists(os.path.dirname(html_page)):
+                os.makedirs(os.path.dirname(html_page))
+            with open(html_page, 'w') as html_file:
+                html_file.write(HTML_HEAD_OPEN)
+                html_file.write("\t\t<style>\n")
+                html_file.write(SAMPLE_TABLE_STYLE)
+                html_file.write("\t\t</style>\n")
+                html_file.write(create_navbar(all_samples, reports_dir))
+                html_file.write(HTML_HEAD_CLOSE)
+                html_file.write("\t\t<h4>{}</h4>\n".format(str(sample_name)))
+                if single_sample.empty:
+                    # When there is no objects.tsv file, search for the
+                    # presence of log, profile, and command files
+                    log_name = os.path.basename(str(glob.glob(os.path.join(
+                                    self.prj.metadata.results_subdir,
+                                    sample_name, '*log.md'))[0]))
+                    profile_name = os.path.basename(str(glob.glob(os.path.join(
+                                        self.prj.metadata.results_subdir,
+                                        sample_name, '*profile.tsv'))[0]))
+                    command_name = os.path.basename(str(glob.glob(os.path.join(
+                                        self.prj.metadata.results_subdir,
+                                        sample_name, '*commands.sh'))[0]))
+                else:
+                    log_name = str(single_sample.iloc[0]['annotation']) + "_log.md"
+                    profile_name = str(single_sample.iloc[0]['annotation']) + "_profile.tsv"
+                    command_name = str(single_sample.iloc[0]['annotation']) + "_commands.sh"
+                # Get relative path to the log file
+                log_file = os.path.join(self.prj.metadata.results_subdir,
+                                        sample_name, log_name)
+                log_relpath = os.path.relpath(log_file, reports_dir)
+                # Grab the status flag for the current sample
+                flag = glob.glob(os.path.join(self.prj.metadata.results_subdir,
```

Comment:
This looks like an assumption that there will always be exactly one flag, or that if there's more than one, we're uninterested in what it is. Is that the case? What's the basis for the guarantee of exactly one flag? Is it just confidence in our own management of the state of a sample's folder?

Reply:
well... in theory, at least with pypiper pipeilnes, there should only ever be one flag. this hasn't always been the case, but recent version fo pypiper delete existing flags when they start.

Reply:
Is the consensus to rework this to handle multiple flags, or to maintain the assumption of a single?

Reply:
I think it's your call. I mostly just wanted to see what the thinking about the state of a sample's flags was. I think it'd be nice to either:
1. Wrap this in either a `try/except` to catch `IndexError` and provide an explanation about a missing flag file to protect against a corner case in which none exists
or
2. Wrap in conditional logic, raise an exception or warning for 0 flags case and/or multiple flags case

But I don't have a strong preference. Regardless I think this is an unlikely scenario so not a big deal if it remains as is, but I think it could be nice to have some logic in place for this while it's being considered to avoid potentially more time consuming diagnostics if something did arise.

Reply:
Addressed this with fe54eec.  Will throw warnings for zero or multiple flags but doesn't break anything.

Reply:
Nice! 👍 
Got tired of looking for a fuzzy blanket emjoi, but this feels nice & safe.